backend/src/api/difficulty-adjustment.ts

```diff
@@ -6,33 +6,37 @@ class DifficultyAdjustmentApi {
   constructor() { }
 
   public getDifficultyAdjustment(): IDifficultyAdjustment {
+    const ESTIMATE_LAG_BLOCKS = 146; // For first 7.2% of epoch, don't estimate.
+    const EPOCH_BLOCK_LENGTH = 2016;
+
     const DATime = blocks.getLastDifficultyAdjustmentTime();
     const previousRetarget = blocks.getPreviousDifficultyRetarget();
     const blockHeight = blocks.getCurrentBlockHeight();
     const blocksCache = blocks.getBlocks();
     const latestBlock = blocksCache[blocksCache.length - 1];
 
-    const now = new Date().getTime() / 1000;
-    const diff = now - DATime;
-    const blocksInEpoch = blockHeight % 2016;
-    const progressPercent = (blocksInEpoch >= 0) ? blocksInEpoch / 2016 * 100 : 100;
```

Comment:
Q: All of these blocksInEpoch >= 0 checks are assuming blockHeight is possibly negative. If blockHeight is negative, nextRetargetHeight and remainingBlocks will be messed up. Are these values ok for such a messed up state?

Reply:
No opinions?

Closing for now, as it shouldn't be a blocker.